src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2909,6 +2910,137 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
+        Protected Overrides Function CommonCreateBuiltinOperator(
+                name As String,
+                returnType As ITypeSymbol,
+                leftType As ITypeSymbol,
+                rightType As ITypeSymbol) As IMethodSymbol
+
+            Dim vbReturnType = returnType.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(returnType))
+            Dim vbLeftType = leftType.EnsureVbSymbolOrNothing(Of NamedTypeSymbol)(NameOf(leftType))
+            Dim vbRightType = rightType.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(rightType))
+
+            Dim nameToCheck = name
+            Select Case name
+                Case WellKnownMemberNames.CheckedAdditionOperatorName
+                    nameToCheck = WellKnownMemberNames.AdditionOperatorName
+                Case WellKnownMemberNames.CheckedDivisionOperatorName
+                    nameToCheck = WellKnownMemberNames.IntegerDivisionOperatorName
+                Case WellKnownMemberNames.CheckedMultiplyOperatorName
+                    nameToCheck = WellKnownMemberNames.MultiplyOperatorName
+                Case WellKnownMemberNames.CheckedSubtractionOperatorName
+                    nameToCheck = WellKnownMemberNames.SubtractionOperatorName
+            End Select
+
+            Dim opInfo = OverloadResolution.GetOperatorInfo(nameToCheck)
+            If Not opInfo.IsBinary Then
+                Throw New ArgumentException($"Illegal operator name '{name}'", NameOf(name))
+            End If
+
+            CheckBinaryBuiltInOperator(name, vbReturnType, vbLeftType, vbRightType, opInfo)
+
+            Return New SynthesizedIntrinsicOperatorSymbol(vbLeftType, name, vbRightType, vbReturnType)
+        End Function
+
+        Private Shared Sub CheckBinaryBuiltInOperator(
+                name As String,
+                returnType As TypeSymbol,
+                leftType As NamedTypeSymbol,
+                rightType As TypeSymbol,
+                opInfo As OverloadResolution.OperatorInfo)
+
+            ' Built in enum binary operators
+            If leftType.IsEnumType() AndAlso
+               leftType.Equals(rightType, TypeCompareKind.ConsiderEverything) AndAlso
+               leftType.Equals(returnType, TypeCompareKind.ConsiderEverything) Then
+                If opInfo.BinaryOperatorKind = BinaryOperatorKind.Xor OrElse
+                   opInfo.BinaryOperatorKind = BinaryOperatorKind.And OrElse
+                   opInfo.BinaryOperatorKind = BinaryOperatorKind.Or Then
+                    Return
+                End If
+            End If
+
+            ' Quick table access to determine if these types are legal.
+            Dim resolved = OverloadResolution.ResolveNotLiftedIntrinsicBinaryOperator(opInfo.BinaryOperatorKind, leftType.SpecialType, rightType.SpecialType)
+            If resolved <> SpecialType.None Then
+                ' Quick access table strangely maps `string Like string` to the `string` return type. remap it to 'bool'
```

Comment:
>Quick access table strangely maps `string Like string` to the `string` return type. remap it to 'bool' [](http://example.com/codeflow?start=18&length=102)

I think it actually maps it to the operand type, which for most operators is equal to return type. However, there are outliers.

Reply:
ok.  i misunderstood teh table then.  have adjusted the callsite accordingly.   will see if tests reveal any other things.

Reply:
>  have adjusted the callsite accordingly.

Not sure if any adjustments were necessary apart from comments, perhaps